[16.0][IMP] sale_order_safe_commitment_date: fill delivery date

chienandalu opened the pull request:
The delivery date is computed by default from the expected date. We can still edit it, but if can't be delivered, it will be force on confirmation.

cc @moduon @rafaelbn @Gelojr  @Shide   MT-9790

chienandalu committed:
[IMP] sale_order_safe_commitment_date: fill delivery date

The delivery date is computed by default from the expected date.
We can still edit it, but if can't be delivered, it will be force
on confirmation.

MT-9790

EmilioPascual reviewed (approved):
Functional and code review

https://www.loom.com/share/0804b4a5f18842caa0ef0863ba1bf4ab?sid=1f53f8aa-7372-480e-ae46-78d500e7eda9

EmilioPascual committed:
[BUILD] sale_order_safe_commitment_date: mark as rebel

sale_order_safe_commitment_date breaks test of other modules, so it's better split to it

@moduon MT-9790

rafaelbn reviewed (approved):
Thank you!

Reviewed, this improvement will help a lot! 👍🏼 😄

rafaelbn commented:
/ocabot merge minor
